rust/private/rustdoc.bzl

```diff
@@ -85,6 +85,9 @@ def _rust_doc_impl(ctx):
     args.add(crate.root.path)
     args.add("--crate-name", crate.name)
     args.add("--crate-type", crate.type)
+    if crate.type == "proc-macro":
```

Comment:
Changes in this and the rustdoc_test files are not really relevant to the rest of the PR. I'd prefer splitting these changes into a separate PR with a focused description, and a unit test similar to https://github.com/bazelbuild/rules_rust/blob/main/test/unit/use_libtest_harness/use_libtest_harness_test.bzl. Wdyt?

Reply:
It's the same core bug though. I don't think a separate PR should be needed for fixing the same class of problem in different rules. If the tech-debt were solved here to have the rules use the same functionality as the core rules for generating flags, there would have been no change to these rules. I think think it's appropriate to keep these changes in this PR and bump the priority on addressing tech debt. Thoughts?

Reply:
Also, the functionality here is covered by: 
https://github.com/bazelbuild/rules_rust/blob/d31b77510173f65356a1b56452d7f2ed1d1a68de/examples/proc_macro/BUILD.bazel#L39-L47
so it's not entirely unchecked.

Reply:
Though I suppose I only mention `rust_test` in the description....

Reply:
Yes you are right. I'd still write an unit test, but you can carry on :)